helm/external-dns-app/templates/deployment.yaml

```diff
@@ -54,11 +69,17 @@ spec:
         {{- if eq .Values.provider "azure" }}
         - --azure-config-file=/azure/config/azure.yaml
         {{- end }}
+        {{- if eq .Values.aws.access "external" }}
+        - --domain-filter={{ .Values.aws.baseDomain }}
+        - --aws-zone-type=public
+        {{- else }}
         - --domain-filter={{ .Values.baseDomain }}
+        {{- end }}
         - --metrics-address=:{{ .Values.metricsPort }}
         - --namespace={{ .Values.namespace }}
         - --registry=txt
         - --txt-owner-id=giantswarm-io-external-dns
+        - --txt-prefix={{ .Values.clusterID }}
```

Comment:
It seems this was a breaking change. See https://github.com/kubernetes-sigs/external-dns#note

It was uncovered while rolling out migration of nginx LB Service management from azure-operator to nginx app. During migration there are two nginx LB Services present, and a migration script is used to switch DNS records to the new LB IP. It could hang, switch would never complete (would continue to resolve to old LBs IP), external-dns doesn't automatically migrate records due to broken ownership.

We're updating the migration process steps https://github.com/giantswarm/releases/pull/330

Still, since issue of broken records ownership is uncovered, it potentially affects multiple other clusters on different providers. We'll engage all SEs to check all existing clusters, if they are affected (upgraded to a platform release with external-dns-app v1.1.0 where this PR was included) and update the TXT record where necessary, to prevent any future related issues.

/cc @whites11